### PR TITLE
4.1.5 Code optimization

### PR DESCRIPTION
**Pull request by asakhnenko**
## Description of Issue (if Applicable)
I suggest some improvements to the Hybrid Artificial Neural Network section.

## What Changes Were Made?
- [x] Code refactoring
- [x] Changing PyTorch layer implementation to adhere to these suggestions: [PyTorch Docs](https://pytorch.org/docs/stable/notes/extending.html)
- [x] Added visualizations: quantum circuit, MNIST dataset 
- [x] Added ANN performance testing block
- [x] Typos fix

## How Was This Tested?
ANN accuracy of the test dataset

## Anything Else We Should Know 
- [ ] I believe it can be nice to add the "Suggested literature" section at the bottom of the notebook. There are some good overview papers about QML, e.g. *Jacob Biamonte, Peter Wittek, Nicola Pancotti, Patrick Rebentrost, Nathan Wiebe, and Seth Lloyd.  Quantum machine learning, Nature,  Sep 2017.*

### CODE REVIEW

**Comment by frankharkins** [comment]
@asakhnenko thank you for this!
